examples/java/src/main/java/io/scanbot/sdk/ScanbotSDKExample.java

```diff
@@ -160,49 +50,57 @@ public static void main(String[] args) throws Exception {
         // need to notify server you are no longer using the license. Alternatively, you can manually call
         // io.scanbot.sdk.ScanbotSDK.deregisterDevice and io.scanbot.sdk.ScanbotSDK.waitForOnlineLicenseCheckCompletion if you need asynchronous
         // deregistration behaviour
-        try (DeviceSession deviceSession = new DeviceSession(DeviceSession.DEFAULT_CLOSE_TIMEOUT_MS)) {
-            String filePath = null;
-            String resourcePath = null;
-
-            if (args.length < 1) {
-                printUsage();
-                return;
-            }
-
-            Map<String, String> argsMap = new HashMap<>();
-            for (int i = 1; i + 1 < args.length; i += 2) {
-                argsMap.put(args[i], args[i + 1]);
-            }
-
-            if (argsMap.containsKey("--resource")) {
-                resourcePath = argsMap.get("--resource");
-            } else if (argsMap.containsKey("--file")) {
-                filePath = argsMap.get("--file");
-            } else {
-                printUsage();
-                return;
-            }
-
-            String savePath = null;
-            if (argsMap.containsKey("--save")) {
-                savePath = argsMap.get("--save");
-            }
-
-            String command = args[0];
-            switch (command.toLowerCase()) {
-                case "detectdocument":
-                    detectDocument(filePath, resourcePath);
+        try (DeviceSession ignored = new DeviceSession(DeviceSession.DEFAULT_CLOSE_TIMEOUT_MS)) {
+            switch (category) {
+                case "scan": {
+                    if (file == null && resource == null) { ExampleUsage.print(); return; }
+                    try (ImageRef image = Utils.createImageRef(file, resource)) {
```

Comment:
We can add calls to ImageRefProfiler after work is done to check if we properly set all the try-with-resources